### ISSUE TITLE
Make thumbnails bigger and easier to distinguish

### ISSUE THREAD

**Issue by HubKing** [s-leger/archipack#161]
Archipack version: 1.2.83

Blender version: 2.81a

Os version: Windows 10

Description of the issue: I tried to add a window, but the thumbnails were too small and had little contrast that it was difficult to see what they look like. Maybe colour the preview models and add some zoom option (like [+] [-] buttons)?

![toosmall](https://user-images.githubusercontent.com/10900489/73133422-aad53800-406b-11ea-92a9-2db30dfb3081.PNG)

**Comment by s-leger** [comment]
Hi, 
Download default material library and setup according [this post](https://github.com/s-leger/archipack/wiki/Setup#download-and-setup-default-material-library)  re-render your thumbs   then restart blender.
Icon sizes are limited by blender to 256 pixels, zoom factor for thumbs feature is available in [archipack 2.x series](https://gum.co/ZRMyP).

**Comment by foreachthing** [comment]
> Icon sizes are limited by blender to 256 pixels, zoom factor for thumbs feature is available in [archipack 2.x series](https://gum.co/ZRMyP).

This is no fun an a 4k monitor, you know.... at least make it some percentage of the screen size, pleeeeeeas?!

**Comment by foreachthing** [comment]
@HubKing :

To make those thumbnails bigger, find the file `archipack_preset.py`. Then change in `Class PresetMenu()` the line containing `self.thumbsize=` to `self.thumbsize=Vector((320, 240))`. Or you can use any other values until you're happy.

To render the preview in a higher resolution, find and edit `archipack_thumbs.py`. Then find `engine settings` and edit to
```
# engine settings
render.resolution_x = 320
render.resolution_y = 240
```
Or, whatever you like ;-)

Save all files and restart Blender. Re-render thumbnails.

**Comment by s-leger** [comment]
Not certain rendering bigger thumbs will do anything, if i recall correctly thumbs size are hardcoded in blender make it impossible to load more than 320x240 pixels. (maybe api changed in the meantime)
But thumbs size (upscaling) in gl display is possible with the .thumbsize trick.

**Comment by foreachthing** [comment]
No, you're right. Rendering bigger thumbs does nothing. Except for improving quality if one would adjust `self.thumbsize=Vector((320, 240))` to a higher resolution.
I rendered my thumbs at 640x480 and added a setting to the addon preferences. Now I can change the thumbnail (display!) size in there.
I got a 4k monitor at work, so I needed bigger thumbnails... ;-)

**Comment by s-leger** [comment]
Rendering more than 320x240 is useless unless the api changed in the meantime, it will not increase quality as loaded thumb size still is 320x240 (this is a blender's limitation).
It also may quickly become an issue when loading thumbs, as you may end up with 100 thumbs and load time no more be in acceptable range (not event talking about memory management).

Also, feel free to push your improvements right into blender's git repository so it can benefit to every single user.

**Comment by foreachthing** [comment]
> Rendering more than 320x240 is useless unless the api changed in the meantime,...

I need to read into that first, before I PR anything...

>It also may quickly become an issue when loading thumbs, as you may end up with 100 thumbs and load time no more be in acceptable range (not event talking about memory management).

Yeah, I understand your point there!
I got some work to do...

**Comment by s-leger** [comment]
From my own experience with pro version, 320x240 should be enough even on 4k screen, given you are able to change final display scale using the self.thumbsize trick.
This will solve both rendering time, loading time / any possible memory issue(s), while providing sufficient information to user to easly select preset.

**Comment by foreachthing** [comment]
So, here's my PR to Blender addons: https://github.com/blender/blender-addons/pull/5
I just added the possibility to change the preview thumb size and left the rendering as it was - as suggested by you.

This should help @HubKing and others on 4k monitors.

**Comment by s-leger** [comment]
Thank you fot this contribution !
